elf_reader.go

```diff
@@ -29,6 +29,12 @@ type elfCode struct {
 	extInfo  *btf.ExtInfos
 }
 
+// isBpfElf checks if the ELF file is for BPF data.
+// Old LLVM versions set e_machine to EM_NONE.
+func isBpfElf(f *elf.File) bool {
```

Comment:
Maybe just move this if into `NewSafeELFFile`? Then `btf.LoadSpecFromReader` also benefits.

Nit: is the extra function necessary?

Reply:
I had the very same thought about putting it into `NewSafeELFFile()`. But this function is also used by functions like `vdsoLinuxVersionCode()` which opens an ELF that does not contain BPF data. So it didn't work out nicely.

Reply:
Ah good point. New constructor like `NewBPFELFFile()`? Or a helper in `internal`

Reply:
Sorry for the delay of my response :pray: 

I have moved the check into `LoadCollectionSpecFromReader()` as it is a one line check and it is the only place where it is used. So maybe `NewBPFELFFile()` might be too much? It think `NewBPFELFFile()` might be worth if the check is used at least twice. please let us know what you think

Reply:
This is fine, thanks!